tunnel/intercept/proxy/proxy.go

```diff
@@ -17,49 +17,69 @@
 package proxy
 
 import (
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"strconv"
+	"strings"
+	"sync"
+
 	"github.com/michaelquigley/pfxlog"
 	"github.com/openziti/foundation/v2/info"
 	"github.com/openziti/foundation/v2/mempool"
+	"github.com/openziti/transport/v2"
 	"github.com/openziti/ziti/tunnel"
 	"github.com/openziti/ziti/tunnel/dns"
 	"github.com/openziti/ziti/tunnel/entities"
 	"github.com/openziti/ziti/tunnel/intercept"
 	"github.com/openziti/ziti/tunnel/udp_vconn"
-	"github.com/pkg/errors"
-	"io"
-	"net"
-	"strconv"
-	"strings"
-	"sync"
 )
 
 type Service struct {
-	Name     string
-	Port     int
-	Protocol intercept.Protocol
-	Closer   io.Closer
+	Name      string
+	Port      int
+	Protocols []intercept.Protocol
+	Binding   string
+	Closers   map[string]io.Closer
 	sync.Mutex
 	TunnelService *entities.Service
 }
 
-func (self *Service) setCloser(c io.Closer) {
+func (self *Service) addCloser(closerType string, c io.Closer) {
 	self.Lock()
 	defer self.Unlock()
-	self.Closer = c
+	self.Closers[closerType] = c
 }
 
 func (self *Service) Stop() error {
 	self.Lock()
 	defer self.Unlock()
-	if self.Closer != nil {
-		return self.Closer.Close()
+
+	logger := pfxlog.Logger().WithField("service", self.Name)
+
+	var errList []error
+	for listenerType, closer := range self.Closers {
+		logger.Infof("closing %s listener for service of type", listenerType)
+		if err := closer.Close(); err != nil {
+			errList = append(errList, err)
+		}
 	}
-	return nil
+	clear(self.Closers)
+	return errors.Join(errList...)
 }
 
 type interceptor struct {
 	interceptIP net.IP
 	services    map[string]*Service
+	lock        sync.Mutex
+}
+
+func NewDelegate() intercept.Interceptor {
+	return &interceptor{
+		interceptIP: net.IPv4zero,
```

Comment:
127.0.0.1 would be a better default. The all zeros address can also sneak in when the interceptor is initialized by the router in proxy mode (`router/xgress_edge_tunnel_v2/tunneler.go), but I realize that would be a change in behavior